Parser/tokenizer.c

```diff
@@ -432,6 +433,15 @@ update_fstring_expr(struct tok_state *tok, char cur)
         case ':':
             if (tok_mode->last_expr_end == -1) {
                 tok_mode->last_expr_end = strlen(tok->start);
+                PyObject *res = PyUnicode_DecodeUTF8(
```

Comment:
Hummm, not sure if this is the place where we should be doing this. Isn't it better to set the metadata when we create the token itself? This way we don't need to pass down the token all the way to here. This function just updates the pointers and at the point when we have the token ready we set the metadata (because the pointers are ready).

Reply:
I think this is also creating a Python object on every new line if I am not mistaken

Reply:
Let me push a commit to see what I mean